File: src/main/java/ml/glassmc/kiln/standard/KilnStandardPlugin.java
package ml.glassmc.kiln.standard;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

import java.io.File;

public class KilnStandardPlugin implements Plugin<Project> {

    private static KilnStandardPlugin instance;

    public static KilnStandardPlugin getInstance() {
        return instance;
    }

    private Project project;

    @Override
    public void apply(Project project) {
        instance = this;
        this.project = project;
    }

    public File getCache() {
        File cache = new File(this.project.getGradle().getGradleUserHomeDir() + "/caches/glass");
        cache.mkdirs();
        return cache;
    }

    public File getLocalRepository() {
        return new File(this.getCache(), "repository");
    }

    public Project getProject() {
        return project;
    }

}

File: src/main/kotlin/ml/glassmc/kiln/standard/KilnStandardPlugin.kt
package ml.glassmc.kiln.standard

import org.gradle.api.Plugin
import org.gradle.api.Project
import java.io.File

class KilnStandardPlugin: Plugin<Project> {

    lateinit var project: Project

    val cache: File
        get() {
            val cache = File("${this.project.gradle.gradleUserHomeDir}/caches/glass")
            cache.mkdirs()
            return cache
        }

    val localRepository: File
        get() {
            return File(cache, "repository")
        }

    override fun apply(project: Project) {
        instance = this
        this.project = project
    }

    companion object {
        lateinit var instance: KilnStandardPlugin
    }

}
